src/CTTConfig.py

```diff
@@ -0,0 +1,163 @@
+
+        job = parser.add_argument_group('Job handling')
+        job.add_argument('--output-dir', default='jobs',
+                         help='Path where the jobs will be stored')
+        job.add_argument('--rootfs-path',
+                         help='Path to the prebuilt rootfs images directory')
+        job.add_argument('--job-name', help='Name of the job')
+
+        job.add_argument('--rootfs', help='Path to your rootfs image')
+        job.add_argument('--kernel', help='Path to your kernel image')
+        job.add_argument('--dtb', help='Path to your dtb')
+        job.add_argument('--dtb-folder', help='Path to your dtb folder')
+        job.add_argument('--modules', help='Path to your modules tar.gz')
+        job.add_argument('-t', '--tests', nargs='+',
+                         help='Tests to run on the board')
+        job.add_argument('-m', '--tests-multinode', nargs='+',
+                         help='Multinode tests to run')
+
+        lava = parser.add_argument_group('LAVA server options')
+        lava.add_argument('--server', help='LAVA server to send results to')
+        lava.add_argument('--username', help='LAVA username')
+        lava.add_argument('--token', help='LAVA token')
+
+        artifacts = parser.add_argument_group('Artifacts options')
+        artifacts.add_argument('--api-token', help='KernelCI API token')
+        artifacts.add_argument('--tree', default='mainline',
+                               help='KernelCI tree to use')
+        artifacts.add_argument('--branch', default='master',
+                               help='KernelCI branch to use')
+        artifacts.add_argument('--defconfigs', nargs='+',
+                               help='KernelCI defconfig to use')
+        artifacts.add_argument('--no-kci', action='store_true',
+                               help='Ignore KernelCI artifacts')
+
+        ssh = parser.add_argument_group('SSH server options')
+        ssh.add_argument('--ssh-server', help='SSH server')
+        ssh.add_argument('--ssh-username', help='SSH username')
+
+        self.__cmdline = vars(parser.parse_args())
+
+    def __validate_config_file(self):
+        if not self.__config.has_section(DEFAULT_SECTION):
+            raise SectionError('Missing %s section' % DEFAULT_SECTION)
+
+    def __validate_cmdline(self):
+        # We can always just list the boards
+        if self.__cmdline['list']:
+            return
+
+        # Validate that we have all the basic options...
+        for option in mandatory_keys:
+            if not self.__contains__(option):
+                raise OptionError('Missing %s option' % option)
+
+        # ... and the options that can only be passed through the command line
+        if self.__cmdline['boards'] is None:
+            raise OptionError('Missing board')
+
+        if (self.__cmdline['tests_multinode'] is None and
+                self.__cmdline['tests'] is None):
+            raise OptionError('Missing test')
```

Comment:
This will break our automated tests. The current behavior, when no test option are given, is to read which test can be run on which board from `boards.py`. If you do that, `--tests` become mandatory, but you need to provide a mechanism to run easily what has been defined in the each board's configuration.